### PR TITLE
Add alt text enhancements

### LINKED ISSUE #1509: More descriptive alt attribute control
Suggestion:

screenshot 2017-06-27 18 27 51

Cc @joen for the patterns and @afercia for the content.

### PR DESCRIPTION
**Pull request by jasmussen**
## Description

Fixes #1509.

This adds help text to the Alt attribute text control in the inspector for images. In doing so it also adds `helpText` as an option to the `TextControl` component. When filled out, an italicized verbose description will be output below the textfield. 

## How Has This Been Tested?

Tested in Chrome, Mac.

## Screenshot

screen shot 2017-11-21 at 13 30 35

## To test

- Insert an image
- Open the block inspector sidebar
- Verify that help text is present

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: blocks/inspector-controls/text-control/index.js [hunk 1]
```diff
@@ -9,12 +9,12 @@ import { withInstanceId } from '@wordpress/components';
 import BaseControl from './../base-control';
 import './style.scss';
 
-function TextControl( { label, value, help, instanceId, onChange, type = 'text', ...props } ) {
+function TextControl( { label, value, help, helpText, instanceId, onChange, type = 'text', ...props } ) {
```
Comment (youknowriad) [inline blocks/inspector-controls/text-control/index.js]: There's already a "help" prop, I think it's the same thing isn't it?

Comment (jasmussen) [reply blocks/inspector-controls/text-control/index.js]: As far as I can tell, the help prop is not visible, and there primarily for screen-readers. The explicit `helpText` I created actually outputs text that's visible below the textfield too.

File: blocks/inspector-controls/text-control/index.js [hunk 2]
```diff
@@ -23,6 +23,9 @@ function TextControl( { label, value, help, instanceId, onChange, type = 'text',
 				aria-describedby={ !! help ? id + '__help' : undefined }
 				{ ...props }
 			/>
+			{ helpText ? (
```
Comment (mtias) [inline blocks/inspector-controls/text-control/index.js]: I think this is already what `help` does in BaseControl: https://github.com/WordPress/gutenberg/blob/master/blocks/inspector-controls/base-control/index.js#L16

File: blocks/inspector-controls/text-control/index.js (hunk 1 above)
Comment (mtias) [reply blocks/inspector-controls/text-control/index.js]: Should we make the other visible? Unless there's a specific reason for it being hidden, it seems like the same kind of value.

Comment (jasmussen) [reply blocks/inspector-controls/text-control/index.js]: Doh, how did I miss that, it looks like you're right. I'm going to take another stab at that.

File: blocks/inspector-controls/text-control/index.js (hunk 2 above)
Comment (jasmussen) [reply blocks/inspector-controls/text-control/index.js]: Doh, how did I miss that, it looks like you're right. I'm going to take another stab at that.

**Comment by jasmussen** [comment]
Removed the `helpText` and used `help` instead. Good catch. Some of this stuff is magic.

**Review by youknowriad — APPROVED** [review]
👍

**Comment by afercia** [comment]
Well done, thanks 🎉

**Pull request merged**